action.yml

```diff
@@ -26,6 +26,14 @@ inputs:
         required: false
         description: 'What type of annotations show. Options: none, all'
         default: all
+    skip-deps:
+        required: false
+        description: 'If true, the action will skip installing the project dependencies'
+        default: 'false'
+    skip-test-script:
+        required: false
+        description: 'If true, the action will skip running the test script'
+        default: 'false'
```

Comment:
I think it would be better to have only one input here: `skip-step`, because your two inputs allow to handle the following cases:

1. Run everything (default case)
2. Skip the installation of dependencies and run only the test script (required for #37)
3. Skip both steps (necessary for #20).
4. Install dependencies and run the test script.

As you can see, the last case is useless - you don't need to install dependencies without calling the test script. I suggest having only one `skip-step` input, which can only handle the cases you need and would be clearer in my opinion. This input should have 3 options:

1. `none` - execute all steps (default case)
2. `install` - skip the step of installing the dependency (#37)
3. `all` - skip all steps (#20)

Therefore, I think that `skip-step` is the best solution to these issues. But this is just my opinion, I would like to hear what you think about this to make an appropriate decision.

Reply:
That makes sense.